### PR TITLE
docs: update contributing guide with latest instructions

### PR DESCRIPTION
**Pull request by brandonroberts**
Updates the contributing instructions and guidelines

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,19 +4,40 @@ We would love for you to contribute to Nx! Read this document to see how to do i
...
+To build the `nx`, and `schematics` packages, run:
+
+```bash
+yarn build
+```
+
+To install the built packages into your `node_modules` folder, run:
+
+```bash
+yarn linknpm
```
Comment (FrozenPandaz) [inline CONTRIBUTING.md]: I could be wrong but... I never do this... I don't think it's necessary?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -26,23 +47,33 @@ To make sure your changes do not break any unit tests, run the following:
...
 
-To make sure you changes do not break any unit tests, run the following:
+To make sure your changes do not break any E2E tests, run:
 
 ```bash
 yarn e2e
 ```
 
-Running e2e tests can take some time, so it is often useful to run a single test. You can do it as follows:
+Running E2E tests can take some time, so it is often useful to run a single test. To run tests against only changed files, run:
```
Comment (FrozenPandaz) [inline CONTRIBUTING.md]:
```suggestion
Running E2E tests can take some time, so it is often useful to run a single test. To run a single suite of tests, run:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -116,27 +148,31 @@ The scope must be one of the following:
...
 
 ## Migrations
 
 Nx allows users to automatically upgrade to the newest version of the package. If you are introducing a change that would require the users to upgrade their workspace, add a migration to `packages/schematics/migrations`.
 
-Migrations are named in the following fashion: `YYYYMMDD-name.ts` (e.g., 20171129-change-schema.ts).
+Migrations are named using version numbers in the following fashion: `update-major-minor-patch/update-major-minor-patch.ts`, for example, `update-8-1-0/update-8-1-0.ts`.
 
-The `yarn nx-migrate` command will run all the migrations after the one encoded in .angular-cli.json.
+The `yarn nx-migrate` command will run all the migrations after the one encoded in `angular.json`.
```
Comment (FrozenPandaz) [inline CONTRIBUTING.md]: We can move this since it is deprecated.

**Comment by brandonroberts** [comment]
@FrozenPandaz feedback changes done

**Review by FrozenPandaz — APPROVED** [review]
LGTM!

**Pull request merged**